sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -198,14 +210,15 @@ public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse
      *     {@link Mono#error(Throwable)}. If any unexpected scenario happens in poll operation, it should be handled by
      *     client library and return a valid {@link PollResponse}. However if poll operation returns
      *     {@link Mono#error(Throwable)}, the {@link Poller} will disregard that and continue to poll.
+     * @param fetchResultOperation The operation to be called to fetch final result after polling has been completed.
      * @param cancelOperation cancel operation if cancellation is supported by the service. It can be {@code null}
      *      which will indicate to the {@link Poller} that cancel operation is not supported by Azure service.
      * @throws IllegalArgumentException if {@code pollInterval} is less than or equal to zero and if
      * {@code pollInterval} or {@code pollOperation} are {@code null}
      */
     public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation,
-                  Consumer<Poller<T>> cancelOperation) {
-        this(pollInterval, pollOperation, null, cancelOperation);
+                  Supplier<Mono<R>> fetchResultOperation, Function<Poller<T, R>, Mono<T>> cancelOperation) {
```

Comment:
I see Poller being returned in sync clients too. Do we need to have Pollers that take Functions/Suppliers that are not `Mono`?

Reply:
nope,
In Sync clients, user will choose to block / block until,  if they want.